specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -266,7 +293,7 @@
             "description": "Recovery Services Vault to be created.",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/Vault"
+              "$ref": "#/definitions/PatchVault"

```

Comment:
Doing this will cause breaking changes. As I mention below, whether or not the new type is necessary is unclear to me.

Reply:
Hi @marstr ,

PatchTrackedResource doesn't have "location" as required property as opposed to its counter part TrackedResource. This was the reason(All the top level properties should be optional) we had to introduce a new model for PATCH call: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}" . This was a swagger violation and hence we updated it. This solution was suggested by @amarzavery

Here is the exact violation you can refer to:

ERROR (PatchBodyParametersSchema/R2016/RPCViolation): Properties of a PATCH request body must not be required. PATCH operation: 'Vaults_Update' Model Definition: 'TrackedResource' Property: 'location'


Also we already have handling for this in our Service code. Hence this won't break anything.

Reply:
> @sonathan says:
Also we have handling for this in our Service code. Hence this won't break anything.

Sorry, I should have been more clear. Changing the type will cause a breaking change for customers consuming some of the generated SDKs, static languages in particular.

Reply:
I just had a long chat about this offline with @amarzavery. We agree that our hands are tied here. While your change will indeed cause breaking changes for some of the client languages, the consequences of not making this change are not any more palatable. I've labeled this PR as containing potential SDK breaks, but I won't have this issue prevent my approval.

Reply:
@marstr Thank-You, Is there any change with respect to this PR required from my end?